### ISSUE TITLE
Automate CC Email to Site & Sub in AMANDA (On PROD)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1877]
### As the Site & Subdivision group of DSD, I need a copy of certain EX permits emailed to my resource account when they are issued. I am currently manually emailing copies of certain EX permits to the Site & Subdivision group of DSD. I'd like the system to take over this function. 

**Stakeholders:** Kim, Paloma
**DTS Manager:** Nate

**Acceptance Criteria:**
- On DS & EX folders (all subtypes)
- Upon folder issuance, the system emails a copy of the permit to sitesubpermits@austintexas.gov 
    - This happens without user interaction
- This feature works when a user issues the permit, and when the permit is issued automatically as part of the post payment procedure. 

-------

**Older Info:**

_Description: 2/28/18 - Maria Becker should be copied on all EX permits except AWU. This may be updated to include DS permits._
_8/3/18 Update - SSI has a group email account now. Send the permits to that email address instead of Maria._
_Request Date: 2018-11-29 16:36:00
Request ID: DTS18-103020
DTS URL: https://atd.knack.com/dts#service-requests/view-issue-details/5c0069f1efe6294ca5e476c6_

*Migrated from [atd-amanda #34](https://github.com/cityofaustin/atd-amanda/issues/34)*

**Comment by amenity** [comment]
Had asked for this a long time ago and then thought it was going to be part of post payment work since the work made it more onerous; not sure if it will be resolved or needs to stay its own issue. 

Paloma has told DSD that she thinks the smarter way for DSD to get this data would be through a Crystal report, but the DSD Site and Subdivision unit doesn't have any IT support.

**Comment by amenity** [comment]
@SurbhiBakshi - please see if Teri P can maybe look into this. Seems like it should be straightforward.

**Comment by amenity** [comment]
Goes with #1520

**Comment by amenity** [comment]
Would want to check with DSD to create a generic account to receive this information; Maria might be in a different role.

**Comment by amenity** [comment]
ROW analyst has to manually send these emails.

**Comment by TracyLinder** [comment]
@Nadin-Nader I'm adding this one to your plate. Can you work w/ stakeholders during sprint 26 to define acceptance criteria please? 

It should be rather simple; what is the email address these should be sent to, should it be a batch job etc.. 

Thank You!

**Comment by TracyLinder** [comment]
I just updated the issue description with the most current user stories. Acceptance criteria is TBD.

**Comment by Nadin-Nader** [comment]
Hi Paloma/Kim,

Hope you two are doing well and staying safe. I was reaching out regarding the request to Automate the CC email to Site & Sub (#1877); to see if we could set up a meeting this week to go over and define the acceptance criteria for this request.

As of right now I have tomorrow 6/10/2020 1-2pm or 3-4pm and Thursday 6/9/2020 from 9am to 1pm available. Please let me know a good time that works for both of you and I will set up the meeting. 

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by Nadin-Nader** [comment]
From: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov> 
Sent: Tuesday, June 9, 2020 3:02 PM
To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
Cc: Linder, Tracy <Tracy.Linder@austintexas.gov>
Subject: RE: Automate CC Email to Site & Sub (#1877)

Hey Nate!

I can do Thursday from 10-11 or 12-1.  It looks like Kim would be available at those times as well.

Thanks!

**Comment by Nadin-Nader** [comment]
From: Nader, Nadin 
Sent: Tuesday, June 9, 2020 4:27 PM
To: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
Cc: Linder, Tracy <Tracy.Linder@austintexas.gov>
Subject: RE: Automate CC Email to Site & Sub (#1877)

Thank you for the fast response. I have set the meeting for Thursday 6/11/2020 from 10am-11am

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by TracyLinder** [comment]
@Nadin-Nader can you please update this w/ your meeting notes?

**Comment by TracyLinder** [comment]
@Nadin-Nader is this ready for acceptance criteria yet?

**Comment by TracyLinder** [comment]
Nate, can you please add notes from your stakeholder meeting and update the description w/ acceptance criteria?

**Comment by Nadin-Nader** [comment]
My notes got misplaced so i had to send an email to Paloma for the information we discussed: 

Hi Paloma,

Hope you are doing well. I was reaching out because unfortunately, I have misplaced my notes on the acceptance criteria, where we were on this request and where we left off on this. I was wondering if you could help me verify the information I did find by giving me the acceptance criteria that you have come up with on this request, The new email address and if you would like this to be a batch or a part of the Issues process (if it is possible). 
At the moment I have as the acceptance criteria:

Copy of EX Permits must be emailed to the resource account
Email must be sent when EX Permit is issued
Email must be sent to Site and Sub group at DSD
New email address to be emailed from: Need new email address if changing
New email for Site and Sub group: Need new email address if changing

Also, I remember you saying that Maria Becker would not be getting the email anymore, can you provide me with the email that will also be getting the permits?

Lastly, if this cannot be put in the Issue process of the permit can a batch be put in its place? If so, how often will this be ran and will there be any verbiage in the email when sent?

Paloma, I am so sorry to ask for this again. Once I get the information I will get this to the Jets or Launch It to work.

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by TracyLinder** [comment]
CC: sitesubpermits@austintexas.gov

**Comment by TracyLinder** [comment]
This looks ready for development.

**Comment by TracyLinder** [comment]
**Status:** Robert to work on during sprint 34.

**Comment by TracyLinder** [comment]
**Status:** No work performed during sprint 34 due to fee update madness. Will continue working during sprint 35.

**Comment by TracyLinder** [comment]
@licarir can you comment with a status update on this please?

**Comment by licarir** [comment]
@TracyLinder Hello Tracy, I have not had time to devote to the ATD work for several weeks now. If this is high on the priority list, I would advise reassigning it to someone else. I have only done research into how to accomplish this but have not done any configuration/programming on this yet.

**Comment by TracyLinder** [comment]
**Status:** Robert has been unable to work on this. Removing it from his queue and putting it back in the backlog.

**Comment by SubhaShroff** [comment]
@Nadin What information(fields) should the email contain?
 Is it going to be in the form of Report sent as attachment or any other format it should be sent in?

**Comment by SubhaShroff** [comment]
**Status:** In ROW Permitting Meeting yesterday(4/20/2021), Paloma confirmed on the following:
What information(fields) should the email contain? Is it going to be in the form of Report sent as attachment or any other format it should be sent in?
**ANSWER:** It is going to be in the PDF report  same as existing ones for EX, RW that are sent to the permits' Applicants/ROW Contractors, upon payment(If 'Activate upon Payment' info is selected yes).

So, currently, I am working on Back Office coding for EX & DS.

I am still waiting - to split the ticket for BO & Portal, if that is considerable.

**Comment by Nadin-Nader** [comment]
@SubhaShroff i have split the card and added the portal portion as a dependent to this one. Its card #5875

**Comment by SubhaShroff** [comment]
@TracyLinder @Nadin-Nader I have submitted DEV deployment request for the commits(31ba61605949, 11848497a84c) - **SCTASK0133843**
Dated - 4/27/2021

**Comment by SubhaShroff** [comment]
**Status:** Just checked, the commits are deployed in DEV. I need to test and update.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder I tested on DEV Amanda Back Office. Following permits are tested by clicking Issue/Approve button.
11900422(DS), 
11900386, 11900385( EX)
We need to check in  sitesubpermits@austintexas.gov email account for these permits emails.

**Comment by Nadin-Nader** [comment]
@SubhaShroff I spoke to Mariya Becker and she only found one of the emails sent to sitesubpermits@austintexas.gov. The one that was found was for the DS folder (11900422(DS)). i attached the permit
[https://app.zenhub.com/files/140626918/eb1a33e6-23be-44ed-b0f5-d91ecbe771e8/download](https://app.zenhub.com/files/140626918/eb1a33e6-23be-44ed-b0f5-d91ecbe771e8/download)

**Comment by SubhaShroff** [comment]
@Nadin-Nader I tested another set of permits today on DEV. I checked the batch jobs to send the emails to sitesubpermits@austintexas.gov, it all looks good and completed, successfully. So, the emails must be sent this time.
- 11777772 EX
- 11895603 EX austin energy
- 11900880 DS COM driveway - demo
Please check with Mariya Becker.

**Comment by Nadin-Nader** [comment]
@SubhaShroff spoke to Mariya and she sent me the emails that she received from 11777772 EX, 11895603 EX Austin energy, 11900880 DS COM driveway.

She only received 11900880, DS COM driveway, 11777772 EX but did NOT get anything for 11895603 E Austin Energy. Below are the automated messages from the emails and the PDFS are attached:

-----Original Message-----
From: noreply@cityofaustin.org [mailto:noreply@cityofaustin.org]
Sent: Tuesday, May 4, 2021 11:37 AM
To: DSD Site Sub Permits MB <SiteSubPermits@austintexas.gov>
Subject: [Subtype], [PermitNumber] Permit Approved and Attached

*** External Email - Exercise Caution ***

This permit has been issued and is attached to this email.
Please contact the permit staff you've been working with if you have any questions about your permit.
CAUTION: This email was received at the City of Austin, from an EXTERNAL source. Please use caution when clicking links or opening attachments. If you believe this to be a malicious and/or phishing email, please forward this email to cybersecurity@austintexas.gov.

[https://app.zenhub.com/files/140626918/e54333ed-e866-4d51-a9f7-3a183e172724/download](https://app.zenhub.com/files/140626918/e54333ed-e866-4d51-a9f7-3a183e172724/download)

[https://app.zenhub.com/files/140626918/44f5ad0e-b211-4335-8fc0-5ad6c655f99a/download](https://app.zenhub.com/files/140626918/44f5ad0e-b211-4335-8fc0-5ad6c655f99a/download)

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0136261 is submitted for TEST deployment.

**Comment by SubhaShroff** [comment]
@Nadin-Nader TEST deployment is done and I tested for it on **TEST** instance AMANDA, for the following permits. For each following permit, I checked the batch job instance(to send email to SiteSubPermits@austintexas.gov)and those were all successfully completed.

12657567 EX- Austin Energy (AE)
12640803 EX- Austin Transportation (ATD)
12657632 EX- Project Connect

12655480 DS - Com. Curb, Gutter
12451793 DS - Com. Driveway & Sidewalk
12651354 DS - Res. Driveway, Sidewalk, Curb, Gutter

Please check with Mariya Becker, if she received email on SiteSubPermits@austintexas.gov for the listed permits.

**Comment by Nadin-Nader** [comment]
Sent an message to Mariya Becker to look for the emails for the folder numbers below:
12657567 EX- Austin Energy (AE)
12640803 EX- Austin Transportation (ATD)
12657632 EX- Project Connect

12655480 DS - Com. Curb, Gutter
12451793 DS - Com. Driveway & Sidewalk
12651354 DS - Res. Driveway, Sidewalk, Curb, Gutter

## **Status:**

Waiting on Mariya Becker to check the inbox for the emails from AMANDA

**Comment by Nadin-Nader** [comment]
Reached out to AnnMary Becker (dont know if shes related to Mariya Becker) asking her to heck the inbox for the emails. Waiting on her response in order to start work on the portal side

**Comment by Nadin-Nader** [comment]
@SubhaShroff all of the 6 emails were found in the inbox Per AnnMary Becker. The UAT was successful. Now we can work on #5875. Once the portal side is done, this one can get promoted

**Comment by SubhaShroff** [comment]
### Status:
Waiting on #5875 to be completed. #5875 yet to be deployed in TEST.

**Comment by SubhaShroff** [comment]
### Status:
Waiting on #5875 to be completed.

**Comment by SubhaShroff** [comment]
### Status:
Waiting on #5875 to be completed.

**Comment by SubhaShroff** [comment]
### Status:
Waiting for reply from AnnBecker for #5875.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Since #5875 is approved for PROD, I submitted the PROD promotion request for both - **SCTASK0149358**( 31ba61605949, 11848497a84c)

**Comment by SubhaShroff** [comment]
### Status:
PROD Deployment is done for both #1877 and #5875.

**Comment by Nadin-Nader** [comment]
Sent email to Kim and Paloma letting them know this is on PROD. I also asked them to let me know if i can close the cards tomorrow or early next week. See email below:

> Hi Ladies,
> 
> Hope you are both doing well. I wanted to let you know that the changes to cards 1877 and 5875, Automate CC Email to Site & Sub in AMANDA, Are now in PROD. Well they have been for a few days now. Please let me know how things are going with this and if I should close it tomorrow or early next week.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services

**Comment by SubhaShroff** [comment]
### Status:
Waiting for stakeholder feedback to Close.

**Comment by Nadin-Nader** [comment]
Per Paloma, this can be closed along with 5875
